static/js/hashchange.js

```diff
@@ -136,7 +138,23 @@ function do_hashchange_normal(from_reload) {
         case "#narrow": {
             maybe_hide_recent_topics();
             ui_util.change_tab_to("#message_feed_container");
-            const operators = hash_util.parse_narrow(hash);
+            let operators = undefined;
+            try {
+                // This fails for URLs containing
+                // foo.foo or foo%foo due to our fault in special handling
+                // of such characters when encoding. This can also,
+                // fail independent of our fault, so just tell the user
+                // that the URL is invalid.
+                // TODO: Show possible valid URLs to the user.
+                operators = hash_util.parse_narrow(hash);
```

Comment:
Should we add a comment to the top of `hash_util.parse_narrow` noting that it will throw an exception on invalid input, and callers should catch that? Seems like a nice bit of documentation to do in case someone is tempted to add another call site in the future.

Reply:
Good call, added.